### DISCUSSION TITLE
PSRule help needed

### DISCUSSION THREAD

**Discussion by andywills133** [microsoft/PSRule#1282] [Q&A] [answered]
I am getting a strange issue when using an external parameter file for bicep.  The resource deployment works fine but psrule is bringing up an error:

   ERROR  Failed to expand bicep source 'C:\temp\psrule-poc\bicep\main.bicep'. Exception calling "GetBicepResources" with "3" argument(s): "Unable to expand resources because the 
source file 'C:\temp\psrule-poc\bicep\main.bicep' was not valid. An error occurred evaluating expression '[length(parameters('virtualNetworks'))]' line 61. The parameter named 'virtualNetworks' was not set or a defaultValue was defined."

If I set the virtualNetworks parameters within main.bicep the rules are processed without error and if I run bicep build and process main.json there is  no error.:

vnet-uks-test-poc-001 : Microsoft.Network/virtualNetworks [6/6]

   PASS  Azure.Resource.UseTags (AZR-000166)
   PASS  Azure.VNET.UseNSGs (AZR-000263)
   PASS  Azure.VNET.SingleDNS (AZR-000264)
   PASS  Azure.VNET.LocalDNS (AZR-000265)
   PASS  Azure.VNET.SubnetName (AZR-000267)
   PASS  Azure.VNET.Name (AZR-000268)

**Comment by BernieWhite** [comment, marked as answer]
@andywills133 Have a look at #1256. If this doesn't answer the question let us know.

**Reply by andywills133** [reply]
@BernieWhite Thank you for the reply.  I missed an empty array in the vnet properties.
